Project1/project1/application.py

```diff
@@ -33,31 +33,60 @@ def main():
 
 @app.route("/register", methods=["GET","POST"])
 def register():
+    print("starting", session)
+    if 'username' in session:
+        username = session["username"]
+        return (f"<h1>Logged in as {username}</h1>" + "<b><a href = '/dropsession'>click here to log out</a></b>")
+
     if request.method == "GET":
-        return render_template("registration.html")
+        alert = False
+        return render_template("login.html", alert=alert)
     
     elif request.method == "POST":
-        name = request.form.get("name")
-        username = request.form.get("username")
-        password = request.form.get("password")
-        birthday = request.form.get("birthday")
-        address = request.form.get("address")
-        print(name, username, password, birthday, address)
-        user = Users(name=name, username=username, password=password, birthday=birthday, address=address)
-        try:
-            db.session.add(user)
-            db.session.commit()
-            return (f"User {name} successfully added to database...!")
-        except Exception as e:
-            return ("Exception raised! Operation was unsucessful...!")
-
+        if (request.form["button"] == "register"):
+            return render_template("registration.html")
+        
+        elif (request.form["button"] == "submit"):
+            name = request.form.get("name")
+            username = request.form.get("username")
+            password = request.form.get("password")
+            birthday = request.form.get("birthday")
+            address = request.form.get("address")
+            print(name, username, password, birthday, address)
+            user = Users(name=name, username=username, password=password, birthday=birthday, address=address)
+            try:
+                db.session.add(user)
+                db.session.commit()
+                return (f"User {name} successfully added to database...!")
+            except Exception as e:
```

Comment:
where is this exception returned? can the user see

Reply:
This exception is returned to user display, yes user can see the error message